enhancements/machine-config/mco-state-reporting.md

```diff
@@ -0,0 +1,453 @@
+- Uncordoning
+- Uncordoned
+- Updating node state and metrics
+   		 
+#### Resume phase
+- Start config drift monitor
+- Error states:
+   - update stuck -- specifically degraded and/or failures in the functions this is the phase that happens if we either error in any of the above stages
+   - unavailable nodes... an obstacle has been hit, but its not the MCOs fault
+     - Disk Pressure
+   	 - Unschedulable
+
+
+### Workflow Description
+
+When an upgrade is triggered by there being mismatch between a desired and current config or simply just a new MachineConfig being applied, the MachineConfigStates for a specific pool will report the following processes (roughly)
+
+
+```console
+$ oc get mcs
+NAME             HEALTH    PROGRESS   NODE                          ACTION      ERROR
+upgrade-master   Healthy   Ready      ip-10-0-35-46.ec2.internal    NodeReady   
+upgrade-worker   Healthy   Ready      ip-10-0-21-151.ec2.internal   NodeReady   
+```
+
+```console
+$ oc get mcs
+NAME             HEALTH    PROGRESS             NODE                          ACTION                     ERROR
+upgrade-master   Healthy   UpdatePrepering      ip-10-0-35-46.ec2.internal    ClosingConfigDriftMonitor   
+upgrade-worker   Healthy   Ready                ip-10-0-21-151.ec2.internal   NodeReady   
+```
+
+```console
+$ oc get mcs
+NAME             HEALTH             PROGRESS   NODE                          ACTION                      ERROR
+upgrade-master   UpdateInProgress   Ready      ip-10-0-35-46.ec2.internal    DrainingNode   
+upgrade-worker   Healthy            Ready      ip-10-0-21-151.ec2.internal   NodeReady   
+```
+
+```console
+$ oc get mcs
+NAME             HEALTH             PROGRESS   NODE                          ACTION                      ERROR
+upgrade-master   UpdateInProgress   Ready      ip-10-0-35-46.ec2.internal    ApplyingFileAndOSChanges   
+upgrade-worker   Healthy            Ready      ip-10-0-21-151.ec2.internal   NodeReady   
+```
+
+```console
+$ oc get mcs
+NAME             HEALTH             PROGRESS   NODE                          ACTION                      ERROR
+upgrade-master   UpdatePostAction   Ready      ip-10-0-35-46.ec2.internal    RebootingNode   
+upgrade-worker   Healthy            Ready      ip-10-0-21-151.ec2.internal   NodeReady   
+```
+
+```console
+$ oc get mcs
+NAME             HEALTH             PROGRESS   NODE                          ACTION                      ERROR
+upgrade-master   Resuming           Ready      ip-10-0-35-46.ec2.internal    ResumingCfgDriftMonitor   
+upgrade-worker   Healthy            Ready      ip-10-0-21-151.ec2.internal   NodeReady   
+```
+
+```console
+$ oc get mcs
+NAME             HEALTH             PROGRESS   NODE                          ACTION                      ERROR
+upgrade-master   Updated            Ready      ip-10-0-35-46.ec2.internal    DesiredAndCurrentConfigMatch   
```

Comment:
NodeReady,DesiredAndCurrentConfigMatch doesn't seems like ACTION. Maybe instead of ACTION, this field can be STATUS.